File: 奖金.py
from queue import Queue
n, m = map(int, input().split())
cnt, dist = [0 for _ in range(n + 1)], [0 for _ in range(n + 1)]
g, final = [[] for _ in range(n + 1)], []
for _ in range(m):
    u, v = map(int, input().split())
    g[v].append(u)
    cnt[u] += 1

def topsort():
    q = Queue()
    for i in range(1, n + 1):
        if not cnt[i]:
            q.put(i)
    while q.qsize():
        u = q.get()
        final.append(u)
        for v in g[u]:
            dist[v] = max(dist[v], dist[u] + 1)
            cnt[v] -= 1
            if not cnt[v]:
                q.put(v)
    return len(final) == n

if topsort():
    print(sum(dist[1:]) + len(dist) * 100 - 100)
else:
    print("Poor Xed")

File: 奖金.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

signed main() {
    int n, m, a, b;
    cin >> n >> m;
    vector<int> g[n + 1];
    vector<int> final;
    vector<int> cnt(n + 1), dist(n + 1);
    while (m--) {
        cin >> a >> b;
        g[b].emplace_back(a);
        cnt[a]++;
    }

    function<bool()> topsort = [&]() -> bool {
        queue<int> q;
        for (int i = 1; i <= n; ++i) {
            if (!cnt[i]) {
                q.push(i);
            }
        }
        while (q.size()) {
            int u = q.front();
            final.emplace_back(u);
            q.pop();
            for (int v : g[u]) {
                dist[v] = max(dist[v], dist[u] + 1);
                if (--cnt[v] == 0) {
                    q.push({v});
                }
            }
        }
        return final.size() == n;
    };

    if (topsort()) {
        cout << accumulate(dist.begin() + 1, dist.end(), 0) + 100 * dist.size() - 100;
    } else {
        cout << "Poor Xed";
    }
    return 0;
}
